pymllm/bench_serve.py

```
@@ -0,0 +1,395 @@
+    # Warmup (pays JIT/compile + CUDA-graph capture costs once).
+    if args.warmup > 0:
+        print(f"\n--- Warmup ({args.warmup}) ---")
+        for i in range(args.warmup):
+            wr = stream_request(args.url, args.model,
+                                build_prompt(max(sizes), for_decode=True),
+                                max_tokens=32, temperature=args.temperature)
+            note = " (JIT/capture)" if wr.ttft_s > 5.0 else ""
+            print(f"  warmup {i+1}/{args.warmup}: ttft={wr.ttft_s*1000:.0f}ms{note}")
+
+    stats: List[SizeStats] = []
+    print(f"\n--- Benchmark ---")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)